### ISSUE TITLE
x/tools/go/analysis/passes/loopclosure: handle t.Parallel

### ISSUE THREAD

**Issue by alandonovan** [golang/go#28771]
If a test calls t.Run in a loop, and the t.Run function literal contains a call to t.Parallel, then the t.Run effectively turns from a synchronous call into a go statement.

```
func Test(t *testing.T) {
    for name, value := range tests {
        t.Run(name, func(t *testing.T) {
            t.Parallel()
            ...
            use(value) // oops! read of value races with loop index variable update
        }
    }
}
```

The loopclosure check could recognize this case and treat such t.Run calls equivalent to a go statement.

**Comment by findleyr** [comment]
Thanks for your feature request, @alandonovan. @adonovan and I have recently implemented this, in #55972.

**Comment by adonovan** [comment]
Who says Severance is science fiction?
